--- a/FreeFileSync/Source/afs/gdrive.cpp	2023-03-04 22:29:16.294650425 -0500
+++ b/FreeFileSync/Source/afs/gdrive.cpp	2023-03-04 22:39:55.250127710 -0500
@@ -89,8 +89,26 @@
 const char DB_FILE_DESCR[] = "FreeFileSync";
 const int  DB_FILE_VERSION = 5; //2021-05-15
 
-std::string getGdriveClientId    () { return ""; } // => replace with live credentials
-std::string getGdriveClientSecret() { return ""; } //
+const unsigned char GDRIVE_CLIENT_ID_K1[] = {};
+const unsigned char GDRIVE_CLIENT_ID_K2[] = {};
+
+const unsigned char GDRIVE_CLIENT_SECRET_K1[] = {};
+const unsigned char GDRIVE_CLIENT_SECRET_K2[] = {};
+
+std::string getGdriveClientId() {
+    std::string str;
+    for (unsigned int i = 0; i < (sizeof(GDRIVE_CLIENT_ID_K1)/sizeof(GDRIVE_CLIENT_ID_K1[0])); i++) {
+        str += (char)(GDRIVE_CLIENT_ID_K1[i] ^ GDRIVE_CLIENT_ID_K2[i]);
+    }
+    return str;
+}
+std::string getGdriveClientSecret() {
+    std::string str;
+    for (unsigned int i = 0; i < (sizeof(GDRIVE_CLIENT_SECRET_K1)/sizeof(GDRIVE_CLIENT_SECRET_K1[0])); i++) {
+        str += (char)(GDRIVE_CLIENT_SECRET_K1[i] ^ GDRIVE_CLIENT_SECRET_K2[i]);
+    }
+    return str;
+}
 
 
 
