#
# Always use XDG_CONFIG_HOME, even for root.
#
--- a/zen/sys_info.cpp
+++ b/zen/sys_info.cpp
@@ -257,7 +257,9 @@
 
 Zstring zen::getUserDataPath() //throw FileError
 {
+#if 0
     if (::getuid() != 0) //nofail; non-root
+#endif
         if (const std::optional<Zstring> xdgCfgPath = getEnvironmentVar("XDG_CONFIG_HOME");
             xdgCfgPath&& !xdgCfgPath->empty())
             return *xdgCfgPath;
